### PR TITLE
add templates

### PR DESCRIPTION
**Pull request by plosana**
Removed templates functionality from kickoff script. It loops across the plugins and checks if they have templates. In case they do, their templates are added to a list to be displayed later

### CODE REVIEW

File: scipion/__main__.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def main():
 
     elif mode in MODE_DEMO:
         runScript(join(SCIPION_SCRIPTS, KICKOFF)
-                  + ' '.join(sys.argv[2:] if len(sys.argv) > 2 else ''))
+                  + ' {}'.format(sys.argv[2:] if len(sys.argv) > 2 else ''))
```
Comment (pconesa) [inline scipion/__main__.py]: Any good reason to change this? We all are used to use join for this ....is this better?, recommended?

File: scipion/scripts/kickoff.py [hunk 2]
```diff
@@ -421,33 +422,42 @@ def getTemplate(root):
...
+        templates = []
+        # Check if other plugins have json.templates
+        templateFolder = pw.Config.getExternalJsonTemplates()
+        # Check if other plugins have json.templates
+        domain = pw.Config.getDomain()
+        tempList = TemplateList()
+        for pluginName, plugin in domain.getPlugins().items():
+            tDir = TemplateDirData(plugin)
+            if tDir.path:
+                Plugin.getTemplates(tempList, tDir, pluginName, templateFolder)
```
Comment (pconesa) [inline scipion/scripts/kickoff.py]: we should take the plugin from plugin --> plugin.Plugin.getTemplates() and we do not need any of the parameters. All of them can be inferred within the Plugin context?

File: scipion/__main__.py (hunk 1 above)
Comment (plosana) [reply scipion/__main__.py]: With the .join approach, it did not detect the empty space character between both elements, and with .format it worked fine

File: scipion/scripts/kickoff.py [hunk 3]
```diff
@@ -421,33 +436,47 @@ def getTemplate(root):
...
         for candFile in candidates:
             if os.path.isfile(candFile):
                 templates.append(String(candFile))
             else:
                 print(" > %s file does not exist." % candFile)
+        templates = tl.genFromStrList(templates).templates
     else:
+        # Check if other plugins have json.templates
+        domain = pw.Config.getDomain()
+        tempList = TemplateList()
```
Comment (JorMaister) [inline scipion/scripts/kickoff.py]: Clean code suggestion: t1 (line 442) and tempList (line 454) generates the same object in both if and else ways. It should be generated before the if statement.

**Pull request merged**